File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.js
function solution(rank, attendance) {
    const eligibleStudents = rank
        .map((r, i) => ({ id: i, rank: r, attend: attendance[i] }))
        .filter(student => student.attend)
        .sort((a, b) => a.rank - b.rank);
    
    const a = eligibleStudents[0].id;
    const b = eligibleStudents[1].id;
    const c = eligibleStudents[2].id;
    
    return 10000 * a + 100 * b + c;
}

File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.py
def solution(rank, attendance):
    eligible_students = [
        (r, i) 
        for i, r in enumerate(rank) 
        if attendance[i] == True
    ]

    sorted_students = sorted(eligible_students)
    
    a = sorted_students[0][1]
    b = sorted_students[1][1]
    c = sorted_students[2][1]
    
    return 10000 * a + 100 * b + c